src/coreclr/jit/block.h

```diff
@@ -583,12 +590,46 @@ struct BasicBlock : private LIR::Range
 
     bool IsFirstColdBlock(Compiler* compiler) const;
 
-    /* The following union describes the jump target(s) of this block */
-    union {
-        unsigned    bbJumpOffs; // PC offset (temporary only)
-        BasicBlock* bbJumpDest; // basic block
-        BBswtDesc*  bbJumpSwt;  // switch descriptor
-    };
+    unsigned GetJumpOffs() const
+    {
+        return bbJumpOffs;
+    }
+
+    void SetJumpOffs(unsigned jumpOffs)
+    {
+        bbJumpOffs = jumpOffs;
+    }
+
+    BasicBlock* GetJumpDest() const
+    {
+        return bbJumpDest;
+    }
+
+    void SetJumpDest(BasicBlock* jumpDest)
+    {
+        bbJumpDest = jumpDest;
+    }
+
+    bool JumpsTo(const BasicBlock* jumpDest) const
+    {
+        return (bbJumpDest == jumpDest);
```

Comment:
Same assert here.

I wonder if we should treat switch and non-switch more uniformly. i.e., `JumpsTo` might imply any flow graph control transfer, including switches. Maybe we'll evolve this over time.

Reply:
Asserting bbJumpKind is valid proved to be too aggressive in JumpsTo(), as well. Should I try enabling these asserts in a follow-up PR? I think that refactor will be nontrivial, since we null-check bbJumpDest for non-jump blocks in a bunch of places.

Reply:
A follow-up PR to add asserts is fine.

Reply:
It seems odd that `SetJumpKind` has a `Compiler*` arg but `SetJumpKindAndTarget` doesn't. I would expect the latter to be a logical superset.

If you added a `Compiler*` arg you could then fix the switch version to allocate the `BBswtDesc` automatically, perhaps.